xbmc/platform/darwin/osx/network/NetworkMacOS.mm

```diff
@@ -179,45 +181,47 @@
 std::vector<std::string> CNetworkMacOS::GetNameServers()
 {
   std::vector<std::string> result;
-
-  FILE* pipe = popen("scutil --dns | grep -F \"nameserver\" | uniq | tail -n2", "r");
-
-  if (pipe)
+  @autoreleasepool
   {
-    int fineNoPipe = ::fileno(pipe);
-
-    struct pollfd pollFd;
-    pollFd.fd = fineNoPipe;
-    pollFd.events = POLLIN;
-    pollFd.revents = 0;
+    // Create an SCPreferencesRef object for the system preferences
+    auto preferenceName = CFSTR("GetDNSInfo");
+    SCPreferencesRef preferences = SCPreferencesCreate(nullptr, preferenceName, nullptr);
+    if (!preferences)
+    {
+      CFRelease(preferenceName);
+      CLog::Log(LOGWARNING, "Unable to determine nameservers");
+      return result;
+    }
 
-    int pollResult = poll(&pollFd, 1, 1000); // Poll for 1 second
-    if (pollResult > 0 && (pollFd.revents & POLLIN))
+    // Get the current DNS configuration from system preferences
+    CFDictionaryRef currentDNSConfig =
+        static_cast<CFDictionaryRef>(SCPreferencesGetValue(preferences, kSCPrefNetworkServices));
+    if (currentDNSConfig && CFDictionaryGetCount(currentDNSConfig) > 0)
     {
-      char buffer[256] = {'\0'};
-      if (fgets(buffer, sizeof(buffer), pipe))
+      NSDictionary* networkServices = (__bridge NSDictionary*)currentDNSConfig;
+      for (NSString* serviceKey in networkServices)
```

Comment:
preferred way to iterate over NSDictionary is `enumerate...` method - it provides both key and value in the handler block